src/priceClient/adapters/baseAdapter.ts

```diff
@@ -28,10 +45,18 @@ export class BaseHTTPAdapter {
       params: urlArgs ?? {},
     };
 
-    const result = await axios(url, args).catch((err) => {
-      const errMsg = err instanceof AxiosError ? err.message : err.toString();
-      throw new Error(`${this.name} price lookup failure (${errMsg})`);
-    });
-    return result.data;
+    const errs: string[] = [];
+    let tries = 0;
+    do {
+      try {
+        return (await axios(url, args)).data;
+      } catch (err) {
+        const errMsg = axios.isAxiosError(err) || err instanceof Error ? err.message : "unknown error";
+        errs.push(errMsg);
+        if (++tries <= this.retries) await this.sleep(Math.pow(1.5, tries) * 1000); // simple backoff
+      }
+    } while (tries <= this.retries);
+
+    throw new Error(`${this.name} price lookup failure (${errs.join(", ")})`);
```

Comment:
FWIW, if we bump to TypeScrit 4.9 then it should be possible to tack `errs`  in under the `cause` argument to `Error()`. This might be desirable for nicer logging by upper-layer error handlers. Reference: https://github.com/microsoft/TypeScript/issues/50583.

Reply:
Why not bump it? Does it break anything?

Reply:
> Why not bump it? Does it break anything?

Fair question - was wary of a yak shaving exercise but it turned out to be a small diff, and all our tests pass: https://github.com/across-protocol/sdk-v2/pull/159.

Reply:
Has been bumped now :+1:.

Reply:
FWIW I tried using `Error(..., { cause })`, but ended up in some kind of purgatory.

Per https://stackoverflow.com/a/73394582, it seems like we'd need at least es2022 support in our environment (i.e. specified via `lib: [..., "es2022"]` in `tsconfig.json`.

My env seems to support up to es2020 (tsc v4.9.5, nodejs v18.14.1)...but for the following code, I consistently get killed for trying to pass in an Object as the 2nd argument to Error().

```
>> 59     throw new Error(`${this.name} price lookup failure`, { cause: errs });
```

```
(typescript) Error: across-protocol/sdk-v2/src/priceClient/adapters/baseAdapter.ts(59,58): semantic error TS2554: Expected 0-1 arguments, but got 2.
Error: across-protocol/sdk-v2/src/priceClient/adapters/baseAdapter.ts(59,58): semantic error TS2554: Expected 0-1 arguments, but got 2
```

I've also noticed that I can't specify "es2022" for tsconfig.json compilerOptions.lib:
```
$ tsdx build && husky install
rpt2: config error TS6046: Argument for '--lib' option must be: 'es5', 'es6', 'es2015', 'es7', 'es2016', 'es2017', 'es2018', 'es2019', 'es2020', 'esnext', 'dom', 'dom.iterable', 'webworker', 'webworker.importscripts', 'scripthost', 'es2015.core', 'es2015.collection', 'es2015.generator', 'es2015.iterable', 'es2015.promise', 'es2015.proxy', 'es2015.reflect', 'es2015.symbol', 'es2015.symbol.wellknown', 'es2016.array.include', 'es2017.object', 'es2017.sharedmemory', 'es2017.string', 'es2017.intl', 'es2017.typedarrays', 'es2018.asyncgenerator', 'es2018.asynciterable', 'es2018.intl', 'es2018.promise', 'es2018.regexp', 'es2019.array', 'es2019.object', 'es2019.string', 'es2019.symbol', 'es2020.bigint', 'es2020.promise', 'es2020.string', 'es2020.symbol.wellknown', 'esnext.array', 'esnext.symbol', 'esnext.asynciterable', 'esnext.intl', 'esnext.bigint', 'esnext.string', 'esnext.promise'.
```

Ultimately I was able to chase this down to a package named `rollup-plugin-typescript2`, which is a depdendency of `tsdx`. [It seems like rpt2 doesn't yet permit `es2022` to be explicitly specified in `tsconfig.json`](https://github.com/ezolenko/rollup-plugin-typescript2/issues/435) :disappointed:.

Reply:
See also https://github.com/across-protocol/sdk-v2/pull/157#discussion_r1125470672 for an intermediate conclusion on why we can't use `Error(..., { cause})` yet.